peachjam/templates/peachjam/saved_document/_folders_list.html

```diff
@@ -15,15 +15,56 @@ <h2>{% trans 'Saved documents' %}</h2>
         <h5 class="card-header">
           <i class="bi bi-folder"></i> {{ folder.name }}
         </h5>
+        <div class="card-header">
+          <ul class="nav nav-tabs card-header-tabs">
+            <li class="nav-item">
+              <button class="nav-link active"
+                      id="saved-doc-tab-{{ folder.pk }}"
+                      data-bs-toggle="tab"
+                      data-bs-target="#saved-docs-{{ folder.pk }}"
+                      type="button"
+                      role="tab"
+                      aria-controls="saved-docs-{{ folder.pk }}"
+                      aria-selected="true">
+                {% trans 'Saved documents' %}
+              </button>
+            </li>
+            <li class="nav-item">
+              <button class="nav-link"
+                      id="similar-docs-tab-{{ folder.pk }}"
+                      data-bs-toggle="tab"
+                      data-bs-target="#similar-docs-{{ folder.pk }}"
+                      hx-target="#similar-docs-list-{{ folder.pk }}"
+                      hx-get="{% url 'folder_similar_docs' folder.pk %}"
+                      type="button"
+                      role="tab"
+                      aria-controls="similar-docs-{{ folder.pk }}"
+                      aria-selected="false">
+                {% trans 'Recommended documents' %}
+              </button>
+            </li>
+          </ul>
+        </div>
         <div class="card-body">
-          <div>
-            {% if folder.saved_documents.all %}
-              {% for saved_document in folder.saved_documents.all %}
-                {% include "peachjam/saved_document/_folder_list_item.html" %}
-              {% endfor %}
-            {% else %}
-              <p>{% trans 'No saved documents.' %}</p>
-            {% endif %}
+          <div class="tab-content">
+            <div class="tab-pane fade show active"
+                 id="saved-docs-{{ folder.pk }}"
+                 role="tabpanel"
+                 aria-labelledby="saved-doc-tab-{{ folder.pk }}">
+              {% if folder.saved_documents.all %}
```

Comment:
The .all goes to the db on this line and again on the next. Instead use a for loop and an empty tag at the end.

Reply:
neat, i've never used the {% empty %} tag before